htdocs/core/class/html.formmail.class.php

```diff
@@ -307,7 +305,7 @@ function get_form($addfileaction='addfile',$removefileaction='removefile')
 			if (GETPOST('mode','alpha') == 'init' || (GETPOST('modelmailselected','alpha') && GETPOST('modelmailselected','alpha') != '-1'))
 			{
 				$this->clear_attached_files();
-				if (! empty($arraydefaultmessage['joinfiles']) && is_array($this->param['fileinit']))
+				if (! empty($arraydefaultmessage->joinfiles) && is_array($this->param['fileinit']))
```

Comment:
$arraydefaultmessage is an array, why removing ['joinfiles'] ?

Reply:
Well, I explained it in the bug report. getEmailTemplate is returning an array, as opposed to the rest of FormMail functions which return a ModelMail object. Therefore, I changed getEmailTemplate and the places were it was treated as an array in `html.formmail.class.php`. Please check lines from 995 to 1047